ethereum/forge-test/query/QueryTest.sol

```diff
@@ -120,6 +120,80 @@ library QueryTest {
             _accounts
         );
     }
+        
+    /// @dev buildSolanaPdaRequestBytes builds a sol_pda query request from the specified fields.
+    function buildSolanaPdaRequestBytes(
+        bytes memory _commitment,
+        uint64 _minContextSlot,
+        uint64 _dataSliceOffset,
+        uint64 _dataSliceLength,
+        uint8 _numPdas,
+        bytes memory _pdas // Created with buildSolanaPdaEntry()
+    ) internal pure returns (bytes memory){
+        return abi.encodePacked(
+            uint32(_commitment.length),
+            _commitment,
+            _minContextSlot,
+            _dataSliceOffset,            
+            _dataSliceLength,
+            _numPdas,
+            _pdas
+        );
+    }
+
+    /// @dev buildSolanaPdaEntry builds a PDA entry for a sol_pda query.
+    function buildSolanaPdaEntry(
+        bytes32 _programId,
+        uint8 _numSeeds,
+        bytes memory _seeds // Created with buildSolanaPdaSeedBytes()
+    ) internal pure returns (bytes memory){
+        return abi.encodePacked(
+            _programId,
+            _numSeeds,
+            _seeds
+        );
+    }
+
+    // According to the spec, there may be at most 16 seeds.
+    // https://github.com/gagliardetto/solana-go/blob/6fe3aea02e3660d620433444df033fc3fe6e64c1/keys.go#L559
+    uint public constant SolanaMaxSeeds = 16;
+
+    // According to the spec, a seed may be at most 32 bytes.
+    // https://github.com/gagliardetto/solana-go/blob/6fe3aea02e3660d620433444df033fc3fe6e64c1/keys.go#L557
+    uint public constant SolanaMaxSeedLen = 32;
+
+    // Custom errors
+    error SolanaTooManySeeds();
+    error SolanaSeedTooLong();
+
+    /// @dev buildSolanaPdaSeedBytes packs the seeds for a PDA entry into an array of bytes.
+    function buildSolanaPdaSeedBytes(
+        bytes[] memory _seeds
+    ) internal pure returns (bytes memory){
+        if (_seeds.length > SolanaMaxSeeds) {
+            revert SolanaTooManySeeds();
+        }
+
+        bytes memory result;
+        uint numSeeds = _seeds.length;
+
+        for (uint idx; idx < numSeeds;) {
+            if (_seeds[idx].length > SolanaMaxSeedLen) {
+                revert SolanaSeedTooLong();
+            }
+            result = abi.encodePacked(
+                result,
+                abi.encodePacked(
+                    uint32(_seeds[idx].length),
```

Comment:
Same here, I'd push the length to the stack first and then use in both the length check and in the packing